### PR TITLE
Feature/use maskedstring for env vars

### PR DESCRIPTION
**Pull request by gunther-bachmann**
Please merge refactor pull requests first!

### CODE REVIEW

File: tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/TclParameterGeneratorIntegrationTest.xtend [hunk 1]
```diff
@@ -67,6 +78,17 @@ class TclParameterGeneratorIntegrationTest extends AbstractTclGeneratorIntegrati
...
 			  private DummyFixture dummyFixture = new DummyFixture();
 			  
+			  private MaskingString env_confEnvVar = new MaskingString(System.getenv("confEnvVar"));
+			  
+			  private String env_nonConfEnvVar = System.getenv("nonConfEnvVar");
+			  
+			  @Before
+			  public void checkEnvironmentVariablesOnExistence() throws Exception {
+			    org.junit.Assert.assertNotNull("environment variable 'confEnvVar' must not be null", env_confEnvVar.get());
+			    org.junit.Assert.assertNotNull("environment variable 'nonConfEnvVar' must not be null", env_nonConfEnvVar);
```
Comment (jjelschen) [inline tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/TclParameterGeneratorIntegrationTest.xtend]: Why not import org.junit.Assert.... statically?

Comment (jjelschen) [reply tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/TclParameterGeneratorIntegrationTest.xtend]: I get it, its generated Java code, and the generator produces fully-qualified names so it does not have to be "smart" about import statements.

File: tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/TclTypeUsageComputerTest.xtend [hunk 2]
```diff
@@ -28,6 +28,7 @@ import org.testeditor.tcl.TestStepContext
 import org.testeditor.tcl.dsl.services.TclGrammarAccess
 import org.testeditor.tcl.dsl.tests.TclModelGenerator
 import org.testeditor.tcl.dsl.tests.parser.AbstractParserTest
+import org.testeditor.fixture.core.MaskingString
```
Comment (jjelschen) [inline tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/TclTypeUsageComputerTest.xtend]: Why is this import needed (no other changes in this file, apparently...)?

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend [hunk 3]
```diff
@@ -636,15 +649,18 @@ class TclJvmModelInferrer extends AbstractModelInferrer {
...
 					default : throw new RuntimeException('''Cannot append class = '«it?.class»'. ''')
 				}
 			]
 	}
-	
+
+	private def void appendReporterEnterCall(ITreeAppendable output, SemanticUnit unit, String message) {
+		appendReporterEnterCall(output, unit, message, null)
+	}
 }
```
Comment (jjelschen) [inline tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend]: newline at EOF?

File: tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/validation/TclParameterUsageValidatorTest.xtend [hunk 4]
```diff
@@ -212,6 +212,48 @@ class TclParameterUsageValidatorTest extends AbstractParserTestWithDummyComponen
...
+		]
+		tclModel.addToResourceSet('MyTest.tcl')
+		
+		// when then
+		validator.assertError(tclModel, TEST_STEP, TclValidator.INVALID_TYPED_VAR_DEREF)
+	}
+	
+	@Test
+	def void testPublicEnvVarExpectingString() {
+		// given
```
Comment (jjelschen) [inline tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/validation/TclParameterUsageValidatorTest.xtend]: Consider factoring out the "given" part

File: tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/TclTypeUsageComputerTest.xtend (hunk 2 above)
Comment (gunther-bachmann) [reply tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/TclTypeUsageComputerTest.xtend]: it is not needed, organize import was not executed ...

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend (hunk 3 above)
Comment (gunther-bachmann) [reply tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend]: this happens, if not using emacs!
